### PR TITLE
feat(lib): Introduce @ngrx/schematics

### LINKED ISSUE #53: Implement ngrx rules for @angular-devkit/schematics
Pretty soon here the Angular team will be releasing [Schematics](https://github.com/angular/devkit/tree/master/packages/schematics), a new library for doing code generation and modification. It's the tool Angular CLI will use for generating components, tests, etc.

It's extensible, so we should develop a set of Rules for schematics that handle some of the boilerplate of ngrx. Things like:
- generating reducers and tests
- generating reducers w/ a set of passed actions
- generating CRUD-reducers 
- creating effects 
- etc

This is blocked proper until a release of Schematics, but feedback on what kind of rules would be useful is welcome.

### PR DESCRIPTION
**Pull request by brandonroberts**
Implements ngrx schematics, including:

- Setting up the store root
- Setting up a store feature
- Generating an actions file
- Generating a container component
- Generating a reducer
- Generating an effect
- Generating an entity
- Generating a "feature" (actions, effects, reducers)

Closes #53

### CODE REVIEW

File: modules/schematics/src/action/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
```
Comment (MikeRyanDev) [inline modules/schematics/src/action/index.spec.ts]: Is this copyright declaration intentional?

Comment (brandonroberts) [reply modules/schematics/src/action/index.spec.ts]: Some of the schematics were created from examples in the Angular schematics so I wasn't sure whether to remove them all or leave them.

**Comment by peterbsmith2** [comment]
@brandonroberts This is a big feature for the community. It might be helpful for your sanity to get assistance. Contributors are contributors. So you can let me and the rest of us know what you need, I'm happy to help. schematics is going to be life changing for my company! cc: @zoitsa

**Comment by brandonroberts** [comment]
@peterbsmith2 I'm glad you're excited about schematics! The initial work in this PR is pretty much done pending review. What I'd like to do is get this reviewed and merged in so we can generate some nightly builds and get some people from the community to use it and provide feedback before it's published as official.

If you have anything you'd specifically like to see besides what's listed in the PR, add it to #53.

**Comment by MikeRyanDev** [comment]
This is looking good to me so far. Are we waiting for the tests to turn green before removing the WIP?

**Comment by brandonroberts** [comment]
I'll go ahead and remove the WIP and get the build green so we can merge it.

**Comment by brandonroberts** [comment]
This is ready to land @MikeRyanDev

**Comment by sharikovvladislav** [comment]
Awesome!!!!1

**Comment by peterbsmith2** [comment]
@brandonroberts I will want to tinker with this on nightly-builds. You can let me know the best way to get that start. I can be available for any cooperation you need.

**Pull request merged**